server.js

```diff
@@ -2,22 +2,86 @@ const express = require("express");
 const cors = require("cors");
 
 const app = express();
+app.use(express.urlencoded({ extended: true }));
 
 app.use(cors());
+app.use(express.json());
+
+app.get("/", function (request, response) {
+	response.sendFile(__dirname + "./index.html");
+});
 
 const welcomeMessage = {
-  id: 0,
-  from: "Bart",
-  text: "Welcome to CYF chat system!",
+	id: 0,
+	from: "Bart",
+	text: "Welcome to CYF chat system!",
 };
 
 //This array is our "data store".
 //We will start with one message in the array.
 //Note: messages will be lost when Glitch restarts our server.
+
 const messages = [welcomeMessage];
 
-app.get("/", function (request, response) {
-  response.sendFile(__dirname + "/index.html");
-});
+app.get('/search', (req, res) => {
+	const filteredMessages = messages.filter(mess => mess.text.toLowerCase().includes(req.query.term.toLowerCase()));
+	if (filteredMessages.length > 0) {
+		res.status(200).send(filteredMessages);
+	} else {
+		res.status(400).send('Nothing found');
```

Comment:
Good job catering for this error scenario. I think that the `404` error code would make more sense here
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/404